data/processing.py

```diff
@@ -345,185 +376,348 @@ def load_scan_data(domains):
+    # Customer satisfaction. Parent domains.
+    # print("[%s][%s] Totalling report." % (agency['slug'], 'cust_sat'))
+    eligible = eligible_for('cust_sat', domains, agency)
+    agency['cust_sat'] = {
       'eligible': len(eligible),
-      'participating': 0
+      'participating': len([report for report in eligible if report['participating']])
     }
-    agency_report['participating'] += len([d for d in eligible if
-                                           d['cust_sat']['participating']])
-    print("[%s][%s] Adding report." % (agency['slug'], 'cust_sat'))
-    Agency.add_report(agency['slug'], 'cust_sat', agency_report)
+
+# Create a Report about each tracked stat.
+def full_report(domains, subdomains):
+
+  full = {}
+
+  # HTTPS. Parent and subdomains.
+  print("[https] Totalling full report.")
+  eligible = eligible_for('https', domains) + eligible_for('https', subdomains)
+  full['https'] = total_https_report(eligible)
+
+  print("[crypto] Totalling full report.")
+  eligible = [domain['https'] for name, domain in domains.items() if domain.get('https') and (domain['https'].get('rc4') is not None)]
+  eligible = eligible + [subdomain['https'] for name, subdomain in subdomains.items() if subdomain.get('https') and (subdomain['https'].get('rc4') is not None)]
+  full['crypto'] = total_crypto_report(eligible)
+
+  # Special separate report for preloaded parent domains.
+  # All parent domains, whether they use HTTP or not, are eligible.
+  print("[preloading] Totalling full report.")
+  eligible = [host['https'] for hostname, host in domains.items()]
+  full['preloading'] = total_preloading_report(eligible)
+
+  # Analytics. Parent domains only.
+  print("[analytics] Totalling full report.")
+  eligible = eligible_for('analytics', domains)
+  participating = 0
+  for report in eligible:
+    if report['participating'] == True:
+      participating += 1
+  full['analytics'] = {
+    'eligible': len(eligible),
+    'participating': participating
+  }
+
+
+  # a11y report. Parent domains.
+  # Constructed very differently.
+  print("[a11y] Totalling full report.")
+  eligible_domains = [host for hostname, host in domains.items() if (host.get('a11y') and host['a11y']['eligible'])]
+  full['a11y'] = {}
+  for domain in eligible_domains:
+    full['a11y'][domain['domain']] = domain['a11y']['error_details']
+
+
+  # Customer satisfaction report. Parent domains.
+  print("[cust_sat] Totalling full report.")
+  eligible = eligible_for('cust_sat', domains)
+
+  participating = 0
+  for report in eligible:
+    if report['participating']:
+      participating += 1
+  full['cust_sat'] = {
+    'eligible': len(eligible),
+    'participating': participating
+  }
+
+  return full
 
 
-# TODO: A domain can also be eligible if it has eligible subdomains.
-#       Has display ramifications.
 def eligible_for_https(domain):
-  return (domain["live"] == True)
+  return (
+    (domain["live"] == True) and
+    (domain["branch"] == "executive")
```

Comment:
In theory, we could drop the `domain["branch"] == "executive"` check from the eligibility check, but it doesn't hurt to be explicit.